contracts/MsgOracle.sol

```diff
@@ -0,0 +1,83 @@
+pragma solidity ^0.5.12;
+
+import "node_modules/openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+/**
+@title MsgOracle
+@author Rinke Hendriksen <rinke@ethswarm.org>
+@notice Set the honey prices for various messages in Swarm
+@dev This on-chain msgOracle must be queried by a swarm-node. By reading the event logs, a node is able to construct the following object:
+msgPrices: {TTL, prices: {swarmMsgX: [<validFrom0, price>, <validFromN, price>], swarmMsgX: [<validFrom0, price>, <validFromN, price>]}}.
+this object expires after TTL seconds. The prices object holds all message types with their respective price arrays.
+In the price array of a message, multiple prices can be quoted. The chosen price is the most-recent price.
+*/
+contract MsgOracle is Ownable {
+    /*
+    TTL is the TTL applied by nodes (how long is a query to this oracle valid)
+    oldTTL is used internally when lastUpdated is not longer than oldTTL ago,
+    furthermore, we don't allow updating TTL when lastUpdated is not TTL seconds ago.
+    */
+    uint256 public TTL;
+    uint256 private oldTTL;
+    uint256 private lastUpdated;
+
+    event LogNewTTL(uint256 TTL);
+    event LogSetMsgPrice(bytes32 indexed swarmMsg, uint256 price, uint256 validFrom);
+    event LogRevertMsgPrice(bytes32 indexed swarmMsg, uint256 price, uint256 validFrom);
+    /**
+    @notice Sets an initial value of TTL and sets the owner
+    @dev The owner can be a EOA or a smart-contract (with any arbitrary governance structure) which can call all functions from this contract
+    @param _TTL The initial TTL, effective immediately.
+    */
+    constructor(uint256 _TTL) public Ownable() {
+        TTL = _TTL;
+        lastUpdated = now;
+        emit LogNewTTL(TTL);
+    }
+
+    /**
+    @notice Sets a new value of TTL.
+    @dev The TTL is the duration in seconds of the validity of a query to this oracle.
+    We advise querying the oracle again at a fraction of TTL (e.g. 50%).
+    Sets the TTL from the msgPrices object {TTL, prices}
+    Warning: the owner is responsible for ensuring that we don't update TTL to an unreasonably high value and hereby deadlock the contract for this period.
+    @param _TTL TTL which will be effective after TTL seconds
+    */
+    function newTTL(uint256 _TTL) public onlyOwner {
+        require(lastUpdated + TTL <= now, "MsgOracleOwner: TTL less than TTL seconds ago updated");
```

Comment:
WIthout overflow check this check is meaningless. The caller is in full control of `lastUpdated + TTL`.

Reply:
Note that this TTL is the global TTL and not _TTL. 
Thus, the only way to make this variable overflow is by setting TTL at one point to a very large value.
I agree with adding the overflow check

Reply:
You're right, it's a different variable.